### PR TITLE
Add option for surveys, that a participant can skip questions 

### PR DESCRIPTION
**Pull request by SebastianKapunkt**

### CODE REVIEW

File: threesixty/models.py [hunk 1]
```diff
@@ -16,6 +16,7 @@ class Survey(models.Model):
     employee_gender = models.CharField(_('employee gender'), max_length=6, choices=GENDERS, default='other')
     employee_email = models.EmailField(_('employee email'))
     manager_email = models.EmailField(_('manager email'))
+    participant_can_skip = models.BooleanField('participant can skip', default=True)
```
Comment (codingjoe) [inline threesixty/models.py]: Please add a small help text, that explains something like: `This option allows a participant to skip single questions.`

Comment (codingjoe) [reply threesixty/models.py]: I would also default this to `False`

File: threesixty/static/threesixty/survey.js [hunk 2]
```diff
@@ -54,15 +70,20 @@ class Question {
         this.$no.css('height', distance * 1.5)
       } else if (direction === 'up') {
         this.$yes.css('height', distance * 1.5)
+      } else if (direction === 'right') {
+        this.$skip.css('width', distance * 1.5)
       }
     } else if (phase === 'cancel') {
       this.$yes.height(0)
       this.$no.height(0)
+      this.$no.width(0)
```
Comment (codingjoe) [inline threesixty/static/threesixty/survey.js]: ? should be skip?

File: threesixty/static/threesixty/survey.js [hunk 3]
```diff
@@ -24,28 +27,41 @@ class Question {
...
-    this.checkbox.attr('checked', true)
+    this.select.find('option[value="2"]').attr('selected', 'selected').parent().trigger('change') // 2 = No
     this.$no.height(0)
     this.$yes.animate({height: '100%'}, {duration: 250, start: this.$yes.height(), complete: () => this.form.submit()})
   }
 
+  submitSkip () {
+    this.answered = true
+    this.select.find('option[value="1"]').attr('selected', 'selected').parent().trigger('change') // 1 = Unknown == skip
+    this.$skip.width(0)
```
Comment (codingjoe) [inline threesixty/static/threesixty/survey.js]: should be $no and $yes

File: threesixty/static/threesixty/survey.js [hunk 4]
```diff
@@ -24,28 +27,41 @@ class Question {
...
         }
       }
     })
   }
 
   submitNo () {
     this.answered = true
-    this.checkbox.attr('checked', false)
+    this.select.find('option[value="3"]').attr('selected', 'selected').parent().trigger('change') // 1 = Yes
     this.$yes.height(0)
```
Comment (codingjoe) [inline threesixty/static/threesixty/survey.js]: add skip

File: threesixty/static/threesixty/survey.js [hunk 5]
```diff
@@ -24,28 +27,41 @@ class Question {
...
     this.$no.animate({height: '100%'}, {duration: 250, start: this.$no.height(), complete: () => this.form.submit()})
   }
 
   submitYes () {
     this.answered = true
-    this.checkbox.attr('checked', true)
+    this.select.find('option[value="2"]').attr('selected', 'selected').parent().trigger('change') // 2 = No
     this.$no.height(0)
```
Comment (codingjoe) [inline threesixty/static/threesixty/survey.js]: add skip

**Comment by SebastianKapunkt** [comment]
@codingjoe I think we have to squash all into one. Because as far as I understand the javascript code it depends on the input type checkbox. When changing the field from Boolean to NullBoolean it will change all that to select with options and it wouldn't work. With this it doesn't make sense to have the third commit seperate.

File: threesixty/models.py [hunk 6]
```diff
@@ -18,6 +18,11 @@ class Survey(models.Model):
     manager_email = models.EmailField(_('manager email'))
     is_complete = models.BooleanField(_('is complete'), default=False,
                                       help_text=_('Will show results and reject further answers.'))
+    participant_can_skip = models.BooleanField(
+        'participant can skip',
+        default=True,
```
Comment (codingjoe) [inline threesixty/models.py]: default should be false, to be backwards compatible

Comment (codingjoe) [inline threesixty/models.py]: the field needs to be added to `SurveyCreateView.fields`

**Comment by SebastianKapunkt** [comment]
@codingjoe the checkbox is missing in the create view

**Pull request merged**